encoding/int/encoding.go

```diff
@@ -0,0 +1,42 @@
+package int
+
+import "io"
+
+// Encoder encodes int values.
+type Encoder interface {
+	// Encode a block if ints.
+	Encode(values []int) error
+	// Bytes reutrns the encoded bytes of the encoder.
```

Comment:
This is implementation detail and should not be part of interface definition.

Also typo `reutrns` -> `returns`.

Reply:
Do you want to return bytes in the `Encode` method then?

I thought you mentioned that we should return the encoded bytes in a `Bytes()` call for some reason.

Reply:
Will probably just return `Bytes()` as an `Encode()` return value. Having it as a separate fn call looks like a non-obvious API.

Reply:
👍